src/ray/raylet/scheduling_resources.cc

```diff
@@ -9,10 +9,16 @@ namespace ray {
 
 namespace raylet {
 
+/// Size conversion for internal representation of resource amounts
+constexpr double kResourceConversionFactor = static_cast<double>(1) / 10000;
```

Comment:
Let's get rid of this conversion factor. Instead, the conversion should happen when we "construct" a new `FractionalResourceQuantity` for the first time. Then the `FractionalResourceQuantity` object can have a `ToDouble` method, which we use to populate the flatbuffer object that we send to the worker in `AssignTask`.

And the conversion in the constructor can be done via something like `static_cast<int>(double_quantity * 10000)`.